pkg/solana/config/chain_reader.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import "github.com/smartcontractkit/chainlink-common/pkg/codec"
+
+type ChainReader struct {
+	Namespaces map[string]ChainReaderMethods `json:"namespaces" toml:"namespaces"`
+}
+
+type ChainReaderMethods struct {
+	Methods map[string]ChainDataReader `json:"methods" toml:"methods"`
+}
+
+type ChainDataReader struct {
+	AnchorIDL  string                 `json:"anchorIDL" toml:"anchorIDL"`
+	Procedures []ChainReaderProcedure `json:"procedures" toml:"procedures"`
+}
+
+type ProcedureType int
+
+const (
+	ProcedureTypeInternal ProcedureType = iota
+	ProcedureTypeAnchor
+)
+
+type ChainReaderProcedure chainDataProcedureFields
+
+type chainDataProcedureFields struct {
+	// IDLAccount refers to the account defined in the IDL.
+	IDLAccount string `json:"idlAccount"`
+	// Type describes the procedure type to use such as internal for static values,
```

Comment:
What's an example of internal for static values?  Why wouldn't that just have modifiers to make something hard-coded?

Reply:
So far this is a placeholder. Hard-coding might make sense, but the CLIP introduced the concept of local procedures that don't exactly use RPC calls. I'm not sure what those are yet so I left the option to add them.

Reply:
I think we can remove the comment.  static hard coding was the only concrete reason we introduced it in the CLIP, which the codec now takes care of instead.  I was thinking it might be useful for other things, but since we have no known need for it yet... we can always re-introduce it later if it turns out there is one